From feea5a7962c86625d1cb198dcd2de0bc939728a9 Mon Sep 17 00:00:00 2001
From: Dennis Fink <dennis.fink@c3l.lu>
Date: Wed, 25 Jan 2017 20:51:03 +0100
Subject: [PATCH] Removed System category from desktop entry file. Fixes #11260

---
 mat.desktop | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mat.desktop b/mat.desktop
index 8bb3c06..e669656 100644
--- a/mat.desktop
+++ b/mat.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Categories=Utility;Security;System;
+Categories=Utility;Security;
 Exec=mat-gui
 GenericName=Metadata Anonymisation Toolkit
 Icon=mat
--
libgit2 0.25.0

