diff --git a/dom/plugins/base/nsPluginHost.cpp b/dom/plugins/base/nsPluginHost.cpp
index 7d85376..ced38a2 100644
--- a/dom/plugins/base/nsPluginHost.cpp
+++ b/dom/plugins/base/nsPluginHost.cpp
@@ -1711,6 +1711,8 @@ PRBool nsPluginHost::GhettoBlacklist(nsIFile *pluginFile)
     // libgnashplugin.so, libflashplayer.so, Flash Player-10.4-10.5.plugin,
     // NPSWF32.dll, NPSWF64.dll
     if (strstr(leafStr, "libgnashplugin") == leafStr ||
+        strstr(leafStr, "liblightsparkplugin") == leafStr ||
+        strstr(leafStr, "libfreshwrapper-pepperflash") == leafStr ||
         strstr(leafStr, "libflashplayer") == leafStr ||
         strstr(leafStr, "Flash Player") == leafStr ||
         strstr(leafStr, "NPSWF") == leafStr) {
