From c9be50f968212b01f8d8ad85e59e19c3e67d8578 Mon Sep 17 00:00:00 2001
From: jvoisin <julien.voisin@dustri.org>
Date: Wed, 5 May 2021 20:39:07 +0200
Subject: [PATCH] Fix the CI in Fedora

---
 tests/test_libmat2.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/test_libmat2.py b/tests/test_libmat2.py
index 5b9ad59..b3ba26d 100644
--- a/tests/test_libmat2.py
+++ b/tests/test_libmat2.py
@@ -453,13 +453,17 @@ class TestCleaning(unittest.TestCase):
                 'Encoder':  'HandBrake 0.9.4 2009112300',
             },
             'expected_meta': {
+                'AverageBitrate': 465641,
+                'BufferSize': 0,
                 'CompatibleBrands': ['isom', 'iso2', 'avc1', 'mp41'],
+                'ColorRepresentation': 'nclx 1 1 1',
                 'CompressorID': 'avc1',
                 'GraphicsMode': 'srcCopy',
                 'HandlerDescription': 'SoundHandler',
                 'HandlerType': 'Metadata',
                 'HandlerVendorID': 'Apple',
                 'MajorBrand': 'MP4  Base Media v1 [IS0 14496-12:2003]',
+                'MaxBitrate': 465641,
                 'MediaDataOffset': 48,
                 'MediaDataSize': 379872,
                 'MediaHeaderVersion': 0,
-- 
GitLab

