From de65f4f4d44d7961d1d6cd37b4fe7b5f78c03ea3 Mon Sep 17 00:00:00 2001
From: jvoisin <julien.voisin@dustri.org>
Date: Sun, 9 Sep 2018 19:09:05 +0200
Subject: [PATCH] Improve the resilience of MAT2 wrt. corrupted PNG

---
 libmat2/images.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libmat2/images.py b/libmat2/images.py
index 837f015..03121e1 100644
--- a/libmat2/images.py
+++ b/libmat2/images.py
@@ -62,6 +62,10 @@ class PNGParser(_ImageParser):
 
     def __init__(self, filename):
         super().__init__(filename)
+
+        if imghdr.what(filename) != 'png':
+            raise ValueError
+
         try:  # better fail here than later
             cairo.ImageSurface.create_from_png(self.filename)
         except MemoryError:
-- 
2.18.0
