From 64c73c9854efd531d393e98e92a69085dd22a253 Mon Sep 17 00:00:00 2001
From: meehow <michal@ert.pl>
Date: Fri, 22 Apr 2022 02:39:46 +0200
Subject: [PATCH] gui.move accepts only integers

---
 torbrowser_launcher/__init__.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/torbrowser_launcher/__init__.py b/torbrowser_launcher/__init__.py
index bff3e8e..115fd4f 100644
--- a/torbrowser_launcher/__init__.py
+++ b/torbrowser_launcher/__init__.py
@@ -90,8 +90,8 @@ def main():
     desktop = app.desktop()
     window_size = gui.size()
     gui.move(
-        (desktop.width() - window_size.width()) / 2,
-        (desktop.height() - window_size.height()) / 2,
+        (desktop.width() - window_size.width()) // 2,
+        (desktop.height() - window_size.height()) // 2,
     )
     gui.show()
 
